src/SignalR/clients/csharp/Http.Connections.Client/src/HttpConnection.cs

```diff
@@ -551,14 +551,34 @@ private HttpClient CreateHttpClient()
             httpClient.Timeout = HttpClientTimeout;
 
             // Start with the user agent header
-            httpClient.DefaultRequestHeaders.UserAgent.Add(Constants.UserAgentHeader);
+            httpClient.DefaultRequestHeaders.Add("User-Agent", Constants.UserAgentHeader);
 
             // Apply any headers configured on the HttpConnectionOptions
             if (_httpConnectionOptions?.Headers != null)
             {
                 foreach (var header in _httpConnectionOptions.Headers)
                 {
-                    httpClient.DefaultRequestHeaders.Add(header.Key, header.Value);
```

Comment:
The UserAgent property here expects a `ProductInfoHeaderValue` [info](https://docs.microsoft.com/en-us/dotnet/api/system.net.http.headers.productinfoheadervalue?view=netframework-4.8) which throws with the current User-Agent format we have. So we just set the header the old fashioned way.